index.js

```diff
@@ -1,8 +1,142 @@
+        res.status(400);
+        return res.send('Missing required parameter: jobId');
+    }
+    if (!fs.existsSync(__dirname + '/uploads/')) {
+        fs.mkdirSync(__dirname + '/uploads/');
+    }
+    uploadDirectory = __dirname + '/uploads/' + req.query.jobId;
+    if (!fs.existsSync(uploadDirectory)) {
+        res.status(400);
+        return res.send('No job with that ID.');
+    }
+    if (!fs.existsSync(uploadDirectory + '/ligand_out.pdbqt')) {
+        res.status(200);
+        return res.send('Job still processing.');
+    }
+    var outputPath = uploadDirectory + '/output.zip';
+    var output = fs.createWriteStream(outputPath);
+    var archive = archiver('zip', {
+        zlib: { level: 9 }
+    })
+    output.on('close', function () {
+        var stat = fs.statSync(outputPath);
+        res.writeHead(200, {
+            'Content-Type': 'application/zip',
+            'Content-Length': stat.size
+        });
+        var readStream = fs.createReadStream(outputPath);
+        readStream.pipe(res)
+    })
+    archive.on('error', function (err) {
+        res.status(400);
+        return res.send('File archiving error.');
+    })
+    archive.pipe(output)
+    results = uploadDirectory + '/ligand_out.pdbqt'
+    archive.append(fs.createReadStream(results), { name: 'ligand_out.pdbqt' })
+    log = uploadDirectory + '/log.txt'
+    archive.append(fs.createReadStream(log), { name: 'log.txt' })
+    archive.finalize();
+
+});
+
+app.post('/v1/autodock', (req, res) => {
+    ligand = null;
+    macromolecule = null;
+    fields = {};
+    errorMsg = "";
+    jobId = crypto.createHmac('SHA256', crypto.randomBytes(48))
+    .update(Date.now()
+    .toString())
+    .digest('hex');
+    if (!fs.existsSync(__dirname + '/uploads/')) {
+        fs.mkdirSync(__dirname + '/uploads/');
+    }
+    uploadDirectory = __dirname + '/uploads/' + jobId;
+    if (!fs.existsSync(uploadDirectory)) {
+        fs.mkdirSync(uploadDirectory);
+    }
+    else {
+        res.status(400);
+        errorMsg = 'Hash collision.';
+    }
+    var form = new formidable.IncomingForm();
+    form.multiples = true;
+    form.parse(req);
+    form.on('field', function(name, value) {
+        fields[name] = value
+    })
+    form.on('fileBegin', function (name, file){
+        if (fs.existsSync(uploadDirectory + name + '.pdbqt')) {
+            res.status(400);
+            errorMsg = 'Multiple files with same name uploaded?';
+        }
+        if (name == 'ligand') {
+            ligand = file.name        
+            file.path = uploadDirectory + '/' + 'ligand.pdbqt';
+        }
+        else if (name == 'macromolecule') {
+            macromolecule = file.name
+            file.path = uploadDirectory + '/' + 'macromolecule.pdbqt';
+        }
+        else {
+            res.status(400);
+            errorMsg = 'Unknown file name parameter: ' + name;
+        }
+
+    });
+    form.on('end', function() {
+        if (errorMsg) {
+            res.send(errorMsg);
```

Comment:
This doesn't execution, so the logs get messy with errors about modifying a response after it's been sent